### PR TITLE
Added filtering to passwords and categories by creating a FilterController with a FilterRequest attached to the Controller.

### PR DESCRIPTION
**Pull request by nando1997**

### CODE REVIEW

File: api/app/Http/Controllers/Api/Categories/categoryController.php [hunk 1]
```diff
@@ -8,14 +8,16 @@
 use Illuminate\Support\Facades\Auth;
 use Illuminate\Support\Facades\Validator;
 use App\Http\Resources\CategoryResource;
+use App\Models\PasswordItem;
+use Illuminate\Http\JsonResponse;
+use PgSql\Lob;
```
Comment (bjvanbemmel) [inline api/app/Http/Controllers/Api/Categories/categoryController.php]: Why import this?

File: api/app/Http/Controllers/FilterController.php [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+use App\Models\User;
+use App\Models\Category;
+use Illuminate\Http\JsonResponse;
+use Illuminate\Support\Facades\Password;
+
+class FilterController extends Controller
+{
+    public function filter(FilterRequest $request)
+    {
+
```
Comment (bjvanbemmel) [inline api/app/Http/Controllers/FilterController.php]: Please remove line

**Review by stalina-b — APPROVED** [review]

File: api/app/Http/Requests/FilterRequest.php [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    }
+
+    /**
+     * Get the validation rules that apply to the request.
+     *
+     * @return array<string, \Illuminate\Contracts\Validation\ValidationRule|array|string>
+     */
+    public function rules(): array
+    {
+
```
Comment (bjvanbemmel) [inline api/app/Http/Requests/FilterRequest.php]: Remove empty line

File: api/routes/api.php [hunk 4]
```diff
@@ -60,4 +61,7 @@
 
     //Security check
     Route::post('/security/check', [SecurityCheckController::class, 'checkAllPasswords']);
+
+    // Filter 
+    route::get('/filters', [FilterController::class, 'filter']);
```
Comment (bjvanbemmel) [inline api/routes/api.php]: `route` should be `Route`

**Review by bjvanbemmel — CHANGES_REQUESTED** [review]

**Pull request merged**